packages/fetch/src/index.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  camel,
+  ClientBuilder,
+  ClientDependenciesBuilder,
+  ClientGeneratorsBuilder,
+  generateFormDataAndUrlEncodedFunction,
+  generateVerbImports,
+  GeneratorDependency,
+  GeneratorOptions,
+  GeneratorVerbOptions,
+  GetterPropType,
+  stringify,
+  toObjectString,
+  generateBodyOptions,
+  isObject,
+} from '@orval/core';
+
+const PARAMS_SERIALIZER_DEPENDENCIES: GeneratorDependency[] = [
+  {
+    exports: [
+      {
+        name: 'qs',
+        default: true,
+        values: true,
+        syntheticDefaultImport: true,
+      },
+    ],
+    dependency: 'qs',
+  },
+];
+
+export const getDependencies: ClientDependenciesBuilder = (
+  hasParamsSerializerOptions: boolean,
+) => [...(hasParamsSerializerOptions ? PARAMS_SERIALIZER_DEPENDENCIES : [])];
+
+const generateRequestFunction = (
+  {
+    queryParams,
+    operationName,
+    response,
+    body,
+    props,
+    verb,
+    formData,
+    formUrlEncoded,
+    override,
+  }: GeneratorVerbOptions,
+  { route }: GeneratorOptions,
+) => {
+  const isRequestOptions = override?.requestOptions !== false;
+  const isFormData = override?.formData !== false;
+  const isFormUrlEncoded = override?.formUrlEncoded !== false;
+
+  const getUrlFnName = camel(`get-${operationName}-url`);
+  const getUrlFnProps = toObjectString(
+    props.filter(
+      (prop) =>
+        prop.type === GetterPropType.PARAM ||
+        prop.type === GetterPropType.NAMED_PATH_PARAMS ||
+        prop.type === GetterPropType.QUERY_PARAM,
+    ),
+    'implementation',
+  );
+  const getUrlFnImplementation = `export const ${getUrlFnName} = (${getUrlFnProps}) => {
+${
+  queryParams
+    ? `
+  const normalizedParams = new URLSearchParams();
+
+  Object.entries(params || {}).forEach(([key, value]) => {
+    if (value !== null && value !== undefined) {
+      normalizedParams.append(key, value.toString());
+    }
+  });`
```

Comment:
normalize it by excluding `undefined` and `null`, because `query params` join to `fetch url`.